ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,220 @@
+- Teams register GitOps repo via VS Code plugin, CLI (`konflux register`), or UI
+- Konflux provisions a new Kubernetes namespace and runs initial validation
+- Decouples team onboarding and eliminates complex CODEOWNERS management
+
+### Secrets Management Integration
+
+**Core Capability:**
+- VS Code plugin enables declarative secrets creation directly from the IDE
+- Secrets are managed through Git configuration with secure backend integration
+- Generated secret definitions use External Secrets Operator (ESO) to reference backend stores
+
+**Supported Backends:**
+- **Kubernetes Secrets**: Native Kubernetes secret storage
+- **Vault**: HashiCorp Vault integration for enhanced security features
+- Future backends can be added as needed
+
+**Security Principles:**
+- Plugin only creates secrets, does not read or list existing secrets
+- All secret operations are scoped and audited
+- Optional: CLI fallback (`konflux secrets create`)
+
+### Fleet Awareness
+
+**Multi-Instance Discovery:**
+When a user's organization operates multiple Konflux instances, the VS Code plugin should provide fleet-wide awareness and intelligent instance selection. Fleet metadata is simple metadata with a description of the clusters capabilities and limitations (text), plus an indicator if the cluster has capacity for new tenants namespace. Each Konflux instance should have a complete copy of the fleet metadata.
+
+**Key Capabilities:**
+- **Instance Discovery**: Plugin can discover all Konflux instances within the organization's fleet
+- **Unified Authentication**: Authentication to one instance enables discovery and basic information access across the fleet
+- **Instance Selection Guidance**: Plugin helps users choose the optimal instance for new tenant repo onboarding based on:
+  - Security zones and compliance requirements
+  - Instance capacity and resource availability
+  - Geographic or network proximity
+  - Operational policies (e.g., dev/staging/prod instance separation)
+
+**User Experience:**
+- Plugin presents a unified view of all available instances
+- Provides recommendations for instance selection during tenant onboarding
+- Displays instance health, capacity, and policy information
+- Enables seamless switching between instances for different projects
+
+**Implementation Considerations:**
+- Fleet registry or service discovery mechanism for instance enumeration
+- Federated authentication system across instances
+- Instance metadata API for capacity and policy information
+- Security controls to prevent unauthorized cross-instance access
+
+### CI/Validation Design
+
+- Introduce a Konflux CLI validator (or GitHub Action) for local + CI use
+- Validates schema, references, build logic statically
+- Same validator embedded in VS Code plugin
+- Optional: Plugin lints against live cluster (e.g., available build agents)
+
+### UI Changes
+
+**Preserve:**
+- Dashboard
+- Build and release logs
+- Live environment status
```

Comment:
We deprecated the Environment API a long time ago.

Reply:
removed